app/models/planning_application.rb

```diff
@@ -216,6 +216,20 @@ def parsed_proposal_details
     proposal_details.present? ? JSON.parse(proposal_details) : []
   end
 
+  def proposals_with_metadata
+    parsed_proposal_details.select { |detail| proposal["metadata"].present? }
+  end
+
+  def proposals_with_flags
+    proposals_with_metadata.select { |proposal| proposal["metadata"]["flags"].present? }
+  end
+
+  def flagged_proposal(flag)
+    proposals_with_flags.select do | proposal|
+        proposal["metadata"]["flags"].any? { |element| element == flag }
```

Comment:
indentation

Reply:
If `proposal["metadata"]["flags"].include?(flag)` works, then I think that's a bit more readable

Reply:
It worked!